pkg/infrastructure/services/probe.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
+
+import (
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// HealthCheckProbeType defines the supported probes for the ServiceDefinition
+type HealthCheckProbeType string
+
+const (
+	// QuarkusHealthCheckProbe probe implemented with Quarkus Microprofile Health. See: https://quarkus.io/guides/microprofile-health.
+	// the operator will set the probe to the default path /health/live and /health/ready for liveness and readiness probes, respectively.
+	QuarkusHealthCheckProbe HealthCheckProbeType = "quarkus"
```

Comment:
Do we have a task for Spring Boot support ?
I know that spring boot has something similar as Microprofile Health => https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-features.html
is there any plan to add this to KogitoApp ?

Reply:
Our services for now doesn't have SpringBoot implementation, that's why. For KogitoApp we need to handle the probe customization to the user. Once we migrate the KogitoApp to KogitoBuild and KogitoRuntime, we will have it. Mind opening a JIRA?

Reply:
created https://issues.redhat.com/browse/KOGITO-1488